haystack/preview/dataclasses/chat_message.py

```diff
@@ -0,0 +1,7 @@
+from dataclasses import dataclass
+
+
+@dataclass
+class ChatMessage:
```

Comment:
Do we really need this? Can't we just use a dict as that's what we're actually passing to the OpenAI SDK in the end? See [line 215 of `chatgpt.py`](https://github.com/deepset-ai/haystack/pull/5714/files#diff-0bb109a9201dddfc8426f7ed9a2c65f370cd736c3bec250dde96ecd0037b948dR215).

Also I wouldn't want this to be part of the public API.

Reply:
Yeah it's more of a leftover now that I think about it. Made more sense with the backend

Reply:
Actually I'm going to move it into the `ChatGPTGenerator` module and keep is as private API for now. It's convenient in code and can be reused later when we add components supporting memory.